python/lsst/drp/tasks/metadetection_shear.py

```diff
@@ -0,0 +1,286 @@
+# This file is part of drp_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from lsst.daf.butler import DataCoordinate, DatasetRef
+from lsst.pipe.base.connectionTypes import BaseInput, Output
+
+__all__ = ()
+
+from collections.abc import Collection, Mapping, Sequence
+from typing import Any, ClassVar
+
+import pyarrow as pa
+
+from lsst.pex.config import ListField
+from lsst.pipe.base import (
+    InputQuantizedConnection,
+    NoWorkFound,
+    OutputQuantizedConnection,
+    QuantumContext,
+    PipelineTask,
+    PipelineTaskConfig,
+    PipelineTaskConnections,
+    Struct,
+)
+import lsst.pipe.base.connectionTypes as cT
+from lsst.cell_coadds import MultipleCellCoadd, SingleCellCoadd
+
+
+class MetadetectionShearConnections(PipelineTaskConnections, dimensions={"patch"}):
+    """Definitions of inputs and outputs for MetadetectionShearTask."""
+
+    input_coadds = cT.Input(
+        "DeepCoadd",
+        storageClass="MultipleCellCoadd",
+        doc="Per-band deep coadds.",
+        multiple=True,
+        dimensions={"patch", "band"},
+    )
+
+    # TODO: If there are image-like or other non-catalog output products (e.g.
+    # detection masks), add them here.
+
+    object_catalog = cT.Output(
+        "ShearObject",
```

Comment:
Does this name matter?  If so should this be more specific like "MDetShearObject" ?

Reply:
It does matter, but for at least DR1 I think "ShearObject" makes sense as there will be only one shear catalog table and that's what we've put in the DPDD.  @arunkannawadi may ultimately have been right that we should have put a more specific name in the DPDD, too, but I'm comfortable revisiting that for DR2+ if/when we decide to run another algorithm alongside.

Reply:
I won't be surprised if Xiangchong Li has a code ready by then